poe-tasks/poetry-connector-tasks.toml

```diff
@@ -135,6 +135,55 @@ help = "Get the language of the connector from its metadata.yaml file. Use with
 cmd = """yq eval '.data.connectorBuildOptions.baseImage' $PWD/metadata.yaml"""
 help = "Get the base image of the connector from its metadata.yaml file."
 
+[tool.poe.tasks.get-local-version]
+cmd = """yq eval '.data.dockerImageTag' $PWD/metadata.yaml"""
+help = "Get the version of the connector from its metadata.yaml file. Use with -qq to get just the version."
+
+[tool.poe.tasks.create-connector-registry-artifacts]
+shell = '''
+set -eu
+VERSION=$(poe -qq get-local-version)
+CONNECTOR_NAME=$(poe -qq get-connector-name)
+
+# Create build directory
+mkdir -p "build/releases/${VERSION}"
+
+# Clear existing contents
+rm -rf "build/releases/${VERSION}"/*
+
+# Copy metadata.yaml
+cp metadata.yaml "build/releases/${VERSION}/"
+
+# Copy icon if it exists
+if [ -f icon.svg ]; then
+  cp icon.svg "build/releases/${VERSION}/"
+fi
+
+# Copy Python-specific artifacts based on airbyte-ci analysis
+if [ -f pyproject.toml ]; then
+  cp pyproject.toml "build/releases/${VERSION}/"
+fi
+if [ -f poetry.lock ]; then
+  cp poetry.lock "build/releases/${VERSION}/"
+fi
+if [ -f poetry.toml ]; then
+  cp poetry.toml "build/releases/${VERSION}/"
+fi
+if [ -f setup.py ]; then
+  cp setup.py "build/releases/${VERSION}/"
+fi
+if [ -f requirements.txt ]; then
+  cp requirements.txt "build/releases/${VERSION}/"
+fi
+if [ -f README.md ]; then
+  cp README.md "build/releases/${VERSION}/"
+fi
```

Comment:
This should be a loop or two loops - one with an array of required files and one with an array of files that are optional.